src/psyclone/domain/gocean/nodes/gocean_extract_node.py

```diff
@@ -141,66 +142,111 @@ def create_driver(self, input_list, output_list):
                      funcnames=["PSyDataType"])
         prog.add(use)
 
+        # Use a symbol table to make sure all variable names are unique
+        sym_table = GOSymbolTable()
+        sym = Symbol("PSyDataType")
+        sym_table.add(sym)
+
+        psy_data = sym_table.new_symbol_name("psy_data")
+        sym_table.add(Symbol(psy_data))
         var_decl = TypeDeclGen(prog, datatype="PSyDataType",
-                               entity_decls=["psy_data"])
+                               entity_decls=[psy_data])
         prog.add(var_decl)
 
         call = CallGen(prog,
-                       "psy_data%OpenRead(\"{0}\", \"{1}\")"
-                       .format(module_name, region_name))
+                       "{0}%OpenRead(\"{1}\", \"{2}\")"
+                       .format(psy_data, module_name, region_name))
         prog.add(call)
 
         post_suffix = self._post_name
+
+        # Variables might need to be renamed in order to guarantee unique
+        # variable names in the driver: An example of this would be if the
+        # user code contains a variable 'dx', and the kernel takes a
+        # property 'dx' as well. In the original code that is no problem,
+        # since the property is used via field%grid%dx. But the stand-alone
+        # driver renames field%grid%dx to dx, which can cause a name clash.
+        # Similar problems can exist with any user defined type, since all
+        # user defined types are rewritten to just use the field name.
+        # We use a mapping to support renaming of variables: it takes as
+        # key the variable as used in the original program (e.g. 'dx' from
+        # an expression like field%grid%dx), and maps it to a unique local
+        # name (e.g. dx_0).
+
+        rename_variable = {}
         for var_name in all_vars:
             # TODO #644: we need to identify arrays!!
+            # Support GOcean properties, which are accessed via a
+            # derived type (e.g. 'fld%grid%dx'). In this stand-alone
+            # driver we don't have the derived type, instead we create
+            # variable based on the field in the derived type ('dx'
+            # in the example above), and pass this variable to the
+            # instrumented code.
+            last_percent = var_name.rfind("%")
+            if last_percent > -1:
+                # Strip off the derived type, and only leave the last
+                # field, which is used as the local variable name.
+                local_name = var_name[last_percent+1:]
+            else:
+                # No derived type, so we can just use the
+                # variable name directly in the driver
+                local_name = var_name
+            unique_local_name = sym_table.new_symbol_name(local_name)
+            rename_variable[local_name] = unique_local_name
+            sym_table.add(Symbol(unique_local_name))
```

Comment:
Again, use `name_from_tag`?

Reply:
As above, I need to make sure unique_local_name is not an already existing name, so can't use ``name_from_tag``.

Reply:
I think you've done what I meant rather than what I said - using `new_symbol_name()` is the way to go.